tests/test_check_tx.py

```diff
@@ -485,3 +488,308 @@ def test_check_score(difference, expected_result):
         return_value=10**18 + difference,
     ):
         assert check_score(onchain_data, offchain_data) == expected_result
+
+
+@pytest.mark.parametrize(
+    "scenario,expected_result",
+    [
+        # No hooks defined - should pass
+        ("no_hooks", True),
+        # All hooks executed correctly - should pass
+        ("all_hooks_executed", True),
+        # Missing pre-hook - should fail
+        ("missing_pre_hook", False),
+        # Missing post-hook - should fail
+        ("missing_post_hook", False),
+        # Insufficient gas limit - should fail
+        ("insufficient_gas", False),
+        # First fill with pre-hooks - should pass
+        ("first_fill_with_pre_hooks", True),
+        # Subsequent fill without pre-hooks - should pass
+        ("subsequent_fill_no_pre_hooks", True),
+        # Subsequent fill with pre-hooks present (should ideally fail, but passes because
+        # current implementation doesn't validate that pre-hooks are absent on subsequent fills)
+        ("subsequent_fill_with_pre_hooks", True),
```

Comment:
I disagree, it will always be expected behavior that subsequent filled orders with pre-hooks are allowed to pass.
This is because solvers are always allowed to add arbitrary calls to a settlement. The call then wouldn't count as a hook but just as a custom solver interaction.
Trying to forbid this would be hard: imagine two users who have the same hook (like in the ETH flow, `wrapAll`), the first user has a fill-or-kill order and needs to call that pre-hook, the second user has a partially fillable order that was already partially executed. The two orders can't be executed together because the second order forbids the solver to include the hook for the first.